examples/jetstream/push-consumer/java/Main.java

```diff
@@ -0,0 +1,198 @@
+
+            // Since this is a push consumer, messages will be sent by the server
+            // and pre-buffered by this subscription. We can observe this by using
+            // the `getPendingMessageCount()` method. Messages are buffered asynchronously, so
+            // this pending count may or may not be three.
+            long queuedMsgs = sub.getPendingMessageCount();
+            System.out.printf("%d messages queued\n", queuedMsgs);
+
+            // The maximum number of messages that will be queued is defined by the
+            // `MaxAckPending` option set on a consumer. The default is 1,000.
+            // Let's observe this by publishing a few more events and then check
+            // the pending status again.
+            js.publish("events.4", null);
+            js.publish("events.5", null);
+            js.publish("events.6", null);
+
+            // Let's check if we buffered some more.
+            queuedMsgs = sub.getPendingMessageCount();
+            System.out.printf("%d messages queued\n", queuedMsgs);
+
+            // To *receive* a message, call `nextMsg` with a timeout. The timeout
+            // applies when pending count is zero and the consumer has fully caught
+            // up to the available messages in the stream. If no messages become
+            // available, this call will only block until the timeout.
+            Message msg = sub.nextMessage(Duration.ofSeconds(1));
+            System.out.printf("Received '%s'\n", msg.getSubject());
+
+            // By default, the underlying consumer requires explicit acknowledgements,
+            // otherwise messages will get redelivered.
+            msg.ack();
+
+            // Let's receive and ack another.
+            msg = sub.nextMessage(Duration.ofSeconds(1));
+            System.out.printf("Received '%s'\n", msg.getSubject());
+            msg.ack();
+
+            // Checking out our pending information, we see there are no more
+            // than four remaining.
+            queuedMsgs = sub.getPendingMessageCount();
+            System.out.printf("%d messages queued\n", queuedMsgs);
+
+            // Unsubscribing this subscription will result in the ephemeral consumer
+            // being deleted. Note, even if this is omitted and the process ends
+            // or is interrupted, the server will eventually clean-up the ephemeral
+            // when it determines the subscription is no longer active.
+            sub.unsubscribe();
+
+            // ### Durable (AddConsumer)
+            // An explicit and safer way to create consumers is using `jsm.addOrUpdateConsumer`.
+            // For push consumers, we must provide a `DeliverSubject` which is the
+            // subject messages will be published to (pushed) for a subscription to
+            // receive them.
+            //
+            // Important: such a short AckWait should not be used, it's simply for demonstration purposes in this example.
```

Comment:
Do we need a short ack wait? If you are demonstrating redeliver explain that, otherwise remove ack wait. Again this is the case where you only want to introduce concepts and ideas that you have to.

Reply:
Indeed demonstrating redeliver after AckWait is exceeded. It's not strictly needed to be 1s, just makes the code wait less long. Removed it for now, just using the default AckWait of 30 seconds.